### PR TITLE
libreswan: 3.18 -> 3.29, addressing CVE-2019-12312 & CVE-2019-10155

### PR DESCRIPTION
**Pull request by risicle**
###### Motivation for this change
https://nvd.nist.gov/vuln/detail/CVE-2019-10155
https://nvd.nist.gov/vuln/detail/CVE-2019-12312

This is targeted at staging because of the huge rebuild - however, that rebuild is all due to the small change made to `unbound` required to persuade it to install its libevent-support headers. I haven't tried a full rebuild of these as the addition of this header to the output shouldn't really affect other packages not looking for it. Famous last words. `libreswan` itself has no reverse dependencies.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @afranchuk I am not a user of this package so would appreciate some help in testing.

### CODE REVIEW

File: pkgs/tools/networking/libreswan/default.nix [hunk 1]
```diff
@@ -51,8 +58,6 @@ stdenv.mkDerivation {
     sed -i -e 's|#!/usr/bin/python|#!/usr/bin/env python|' -e 's/^\(\W*\)installstartcheck()/\1sscmd = "ss"\n\0/' programs/verify/verify.in
   '';
 
-  patches = [ ./libreswan-3.18-glibc-2.26.patch ];
```
Comment (ehmry) [inline pkgs/tools/networking/libreswan/default.nix]: Should this patch file be removed from the tree?

**Comment by afranchuk** [comment]
Superficially this looks fine to me, though I can't test anything because I no longer use the package myself...

File: pkgs/tools/networking/libreswan/default.nix (hunk 1 above)
Comment (risicle) [reply pkgs/tools/networking/libreswan/default.nix]: Good catch. Removed.

**Review by ghost — APPROVED** [review]

**Comment by risicle** [comment]
What should we do regarding 19.09? I don't feel comfortable backporting the patch to 3.18 - a lot of code around this area has changed between the two versions.. but it's also quite a big bump. It's "only" a null pointer dereference, but debian seem to have classed it "grave".

**Pull request merged**